src/hotspot/share/prims/jvmtiTagMap.cpp

```diff
@@ -2225,6 +2227,142 @@ class JNILocalRootsClosure : public OopClosure {
   virtual void do_oop(narrowOop* obj_p) { ShouldNotReachHere(); }
 };
 
+// Helper class to collect/report stack roots.
+class StackRootCollector {
+private:
+  JvmtiTagMap* tag_map;
+  JNILocalRootsClosure* blk;
+  // java_thread is needed only to report JNI local on top native frame;
+  // I.e. it's required only for platform/carrier threads or mounted virtual threads.
+  JavaThread* java_thread;
+
+  oop threadObj;
+  jlong thread_tag;
+  jlong tid;
+
+  bool is_top_frame;
+  int depth;
+  frame* last_entry_frame;
```

Comment:
The field names of a helper class are usually started with '_' symbol.

Reply:
renamed all fields